### PR TITLE
Fix build, add math, time, partial signal, and partial dl support.

### PR DESCRIPTION
**Pull request by alexchandel**
This fixes the builds, and adds lots of support. The most significant improvement is the math library.

### CODE REVIEW

File: src/types/linux/mod.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ pub type __kernel_old_uid32_t = __kernel_uid_t;
 pub type __kernel_old_gid32_t = __kernel_gid_t;
 pub type __kernel_old_dev_t   = uint_t;
 pub struct __kernel_fsid_t {
-    pub val: [int_t, ..2],
+    pub val: [int_t; ..2],
```
Comment (mahkoh) [inline src/types/linux/mod.rs]: This syntax looks wrong.

Comment (alexchandel) [inline src/types/linux/mod.rs]: It's not, that's the new array syntax. :\

Comment (mahkoh) [inline src/types/linux/mod.rs]: The new syntax is [T; N] without ..

I just fixed the build. Please rebase.

Comment (mahkoh) [inline src/types/linux/mod.rs]: Oops, that was another library.

Comment (alexchandel) [inline src/types/linux/mod.rs]: Oh I see what you mean. >,< I'll push a fix!

**Comment by alexchandel** [comment]
Pushed fix for linux build.

**Comment by mahkoh** [comment]
I've applied your patches but I had to remove the getenv example because it segfaults on linux.

**Comment by alexchandel** [comment]
Ah that's because I didn't implement crt0 for linux yet. Next time tell me instead, and I'll remove it from my PR.

**Comment by alexchandel** [comment]
What did you do to commit them? By not merging, it seems to have broken my fork.

**Comment by mahkoh** [comment]
```
git am your.patch
# fix the segfault
git commit
git push
```

**Comment by alexchandel** [comment]
Yeah, that renders my fork & local repo incompatible. Next time, can I recommend you merge and then commit your fix on top of it? Or I could just fix it in my PR.

**Comment by mahkoh** [comment]
Have you tried rebasing on master?

**Comment by alexchandel** [comment]
I tried but it was pretty ugly, since master now has half the local changes in a different history. Anyway I'm submitting another PR fixing the linux crt0.

**Pull request closed without merging**